src/Database/Adapter/MariaDB.php

```diff
@@ -608,6 +658,26 @@ public function getSupportForCasting(): bool
         return false;
     }
 
+    /**
+     * Returns the attribute type for read permissions
+     *
+     * @return string
+     */
+    public function getTypeForReadPermission(): string
```

Comment:
```suggestion
    public function getTypeForReadPermission(): string
```

This should be a protected method used internally only by this class.